Library/Homebrew/formulary.rb

```diff
@@ -904,11 +1100,20 @@ def self.path(ref)
 
   def self.tap_formula_name_type(tapped_name, warn:)
     user, repo, name = tapped_name.split("/", 3).map(&:downcase)
-    tap = Tap.fetch user, repo
+    tap = Tap.fetch(user, repo)
     type = nil
 
-    if (possible_alias = tap.alias_table[name].presence)
-      name = possible_alias
+    # FIXME: Remove the need to do this here.
+    alias_table_key = if tap.core_tap?
+      name
+    else
+      "#{tap}/#{name}"
+    end
+
+    if (possible_alias = tap.alias_table[alias_table_key].presence)
+      # FIXME: Remove the need to split the name and instead make
+      #        the alias table only contain short names.
+      name = possible_alias.split("/").last
       type = :alias
```

Comment:
This is related tangentially to https://github.com/Homebrew/brew/issues/16636 and https://github.com/Homebrew/brew/pull/16637.

Reply:
Yes, this is the same fix as in https://github.com/Homebrew/brew/pull/16637.